readme.md

```diff
@@ -47,3 +47,379 @@ See [CONTRIBUTING.md](./CONTRIBUTING.md).
 
 ## What is still left to do?
 See the `TODO.txt` file in this project
+
+---
+
+## 🚀 Comprehensive Guide to Riff-Raff
+
+*"Deploy the transit beam"* – The Guardian's full-stack AWS deployment orchestrator.
+
+### 📚 Table of Contents
+1. [🏗️ Architecture Overview](#️-architecture-overview)
+2. [🧠 Deep Dive: How Riff-Raff Actually Works](#-deep-dive-how-riff-raff-actually-works)
+3. [🗺️ Runtime Architecture](#️-runtime-architecture)
+4. [🔁 CI/CD Pipeline Flow](#-cicd-pipeline-flow)
+5. [📦 Artifact Structure](#-artifact-structure)
+6. [⚙️ Deployment Types](#️-deployment-types)
+7. [🛠️ Configuration with riff-raff.yaml](#️-configuration-with-riff-raffyaml)
+8. [🔗 GitHub Actions Integration](#-github-actions-integration)
+9. [✅ Best Practices](#-best-practices)
+10. [🧪 Troubleshooting](#-troubleshooting)
+11. [🧬 Advanced Features](#-advanced-features)
+12. [📚 Additional Resources](#-additional-resources)
+
+---
+
+### 🏗️ Architecture Overview
+
+Riff-Raff is The Guardian's internally developed deployment platform that acts as a deployment orchestrator (not a CI tool). Written in Scala, it controls all production and staging deployments through a centralized interface and API.
+
+#### 💡 What Riff-Raff Is:
+- 🎯 A centralized service for managing deployments to AWS
+- 🔌 Integrated with GitHub Actions via an API
+- 🛡️ A layer of abstraction over infrastructure provisioning (CDK, CloudFormation)
+- 🚪 A gatekeeper for deployment safety, logging, permissions, and rollback
+
+#### ❌ What Riff-Raff Is Not:
+- 🏗️ A build server (GitHub Actions handles that)
+- 🐳 A container orchestrator (no dynamic compute spin-up)
+- ⚡ Triggered by AWS services (not S3 → EventBridge)
+
+---
+
+### 🧠 Deep Dive: How Riff-Raff Actually Works
+
+#### ✅ CI/CD Division of Labor
+
+| Stage | Responsibility | Who Does It? |
+|-------|---------------|--------------|
+| **CI** | Build, test, upload artifacts | 🔧 GitHub Actions |
+| **Artifact Push** | Send ZIP, CDK, riff-raff.yaml to S3 | 🔧 GitHub Actions |
```

Comment:
we actually build the CDK into CFN in CI and then we upload the CFN JSON files to S3